pkg/parsers/env.go

```diff
@@ -59,5 +59,44 @@ func EnvVarFrom(str string) corev1.EnvVar {
 		}
 	}
 
-	return envvar
+	return envVar
+}
+
+func EnvFromSource(str string) corev1.EnvFromSource {
+	refType := ""
+	refName := ""
+	prefix := ""
+	parts := strings.SplitN(str, ":", 3)
+	switch len(parts) {
+	case 2:
+		prefix = ""
+		refType = parts[0]
+		refName = parts[1]
+	case 3:
+		prefix = parts[0]
+		refType = parts[1]
+		refName = parts[2]
+	}
+
+	envFromSource := corev1.EnvFromSource{}
+	if refType == "secretRef" {
```

Comment:
the `KeyRef` and `Ref` suffix isn't really providing much value here since we have separate flags for each. Can we drop them?

Reply:
it matches the k8s API better, and it makes it more obvious that they are different where KeyRef is "name:key" while Ref is just "name"